File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/buildings/evolution/Evolution.java
package io.github.alexiscomete.lapinousecond.worlds.buildings.evolution;

import io.github.alexiscomete.lapinousecond.worlds.buildings.BuildMethods;
import io.github.alexiscomete.lapinousecond.worlds.buildings.Building;
import io.github.alexiscomete.lapinousecond.worlds.buildings.BuildingInteraction;
import io.github.alexiscomete.lapinousecond.worlds.buildings.Buildings;
import org.json.JSONObject;

public class Evolution {
    private final Buildings evolutionTarget;
    private final double cost;

    public Evolution(Buildings evolutionTarget, double cost) {
        this.evolutionTarget = evolutionTarget;
        this.cost = cost;
    }

    public Evolution(JSONObject jsonObject) {
        this.evolutionTarget = Buildings.valueOf(jsonObject.getString("name").toUpperCase());
        this.cost = jsonObject.getDouble("cost");
    }

    public Buildings getEvolutionTarget() {
        return evolutionTarget;
    }

    public double getCost() {
        return cost;
    }

    public void evolute(Building building) {
        BuildingInteraction buildMethods = evolutionTarget.get(building);
        building.evolute(buildMethods);
        building.set("build_status", "evolute");
        building.set("collect_value", "0.0");
        building.set("collect_target", "0.0");
    }
}

File: src/main/java/worlds/buildings/evolution/Evolution.kt
package worlds.buildings.evolution

import worlds.buildings.Building
import worlds.buildings.Buildings
import org.json.JSONObject
import java.util.*

class Evolution {
    private val evolutionTarget: Buildings
    private val cost: Double

    constructor(evolutionTarget: Buildings, cost: Double) {
        this.evolutionTarget = evolutionTarget
        this.cost = cost
    }

    constructor(jsonObject: JSONObject) {
        evolutionTarget = Buildings.valueOf(jsonObject.getString("name").uppercase(Locale.getDefault()))
        cost = jsonObject.getDouble("cost")
    }

    fun evolute(building: Building) {
        val buildMethods = evolutionTarget[building]
        building.evolute(buildMethods)
        building["build_status"] = "evolute"
        building["collect_value"] = "0.0"
        building["collect_target"] = "0.0"
    }
}